itenium-socks/src/app/socks/shop.component.ts

```diff
@@ -1,21 +1,44 @@
-import { Component } from '@angular/core';
+import {Component, OnInit} from '@angular/core';
 import { SocksService } from './socks.service';
-import { Observable } from 'rxjs';
+import {last, Observable} from 'rxjs';
 import { Sock } from './sock.model';
 import { AsyncPipe, NgFor } from '@angular/common';
+import {MatPaginator, PageEvent} from "@angular/material/paginator";
 
 @Component({
   selector: 'app-shop',
   standalone: true,
-  imports: [NgFor, AsyncPipe],
+  imports: [NgFor, AsyncPipe, MatPaginator],
   templateUrl: './shop.component.html'
 })
-export class ShopComponent {
+export class ShopComponent implements OnInit{
+  totalItems = 100;
+  pageSize = 8;
+  currentPage = 0;
   socks$!: Observable<Sock[]>;
+  socks: Sock[] = [];
```

Comment:
Avoid fields in components.

Het is hier ook niet meer super duidelijk dat `socks` enkel een bepaalde pagina van socks bevat.

Definieer meerdere Observables:

```
socks$!: Observable<Sock[]>;
totalCount$ = this.socks$.pipe(map(socks => socks.length))
filteredSocks = combineLatest([this.socks$, this.pagination$]).pipe(map(([socks, pagination]) => socks.slice(...));
```